lib/stat_tracker.rb

```diff
@@ -22,38 +24,39 @@ def initialize(games_data, teams_data, game_teams_data)
   end
 
   def highest_total_score
-    Game.all.max_by { |game| game.total_score}.total_score
+    @games_data.max_by { |game| game.total_score}.total_score
   end
 
   def lowest_total_score
-    Game.all.min_by { |game| game.total_score}.total_score
+    @games_data.min_by { |game| game.total_score}.total_score
   end
 
   def biggest_blowout
-    Game.all.max_by { |game| game.margin_of_victory }.margin_of_victory
+    @games_data.max_by { |game| game.margin_of_victory }.margin_of_victory
   end
 
   def percentage_home_wins
-    num = Game.all.select{ |game| game.home_goals > game.away_goals}.length
-    denom = Game.all.length
-    (num.to_f / denom * 100).round(2)
+    # Push home wins definition to game data class method & call here?
+    home_wins = @games_data.select{ |game| game.home_goals > game.away_goals}.length
+    average(home_wins, @games_data.length).round(2)
   end
 
   def percentage_visitor_wins
-    num = Game.all.select{ |game| game.away_goals > game.home_goals}.length
-    denom = Game.all.length
-    (num.to_f / denom * 100).round(2)
+    # Push visitor wins definition to game data class method & call here?
+    visitor_wins =  @games_data.select{ |game| game.away_goals > game.home_goals}.length
+    average(visitor_wins, @games_data.length).round(2)
   end
 
   def percentage_ties
-    num = Game.all.select{ |game| game.margin_of_victory == 0}.length
-    denom = Game.all.length
-    (num.to_f / denom * 100).round(2)
+    # Push ties definition to game data class method & call here?
+    ties = @games_data.select{ |game| game.margin_of_victory == 0}.length
+    average(ties, @games_data.length).round(2)
   end
 
   def count_of_games_by_season
-    Game.all.reduce({}) do |games_count, game|
-      if !games_count[game.season.to_s]
+    # Push all this logic to class method in Game & call that here?
+    @games_data.reduce({}) do |games_count, game|
+      if games_count[game.season.to_s].nil?
```

Comment:
This looks great right now. I think when we refactor we will move this logic to a class method on Game.